### PR TITLE
Add `SAVE IMAGE` warning to Earthfile reference

### PR DESCRIPTION
**Pull request by dchw**
A user brought up that this behavior was not well documented. I added it to our documentation for `SAVE IMAGE` to make it more apparent.

### CODE REVIEW

File: docs/earthfile/earthfile.md [hunk 1]
```diff
@@ -677,6 +677,14 @@ SAVE IMAGE my-example-registry.com/another-image:latest
 ```
 {% endhint %}
 
+{% hint style='danger' %}
+##### Important
+
+Since the introduction of the `--referenced-save-only` feature flag, images are only saved if they are connected to the initial target through a chain of `BUILD` commands. [See the `--referenced-save-only` flag documentation for more details](./features.md#referenced-save-only).
```
Comment (alexcb) [inline docs/earthfile/earthfile.md:683]: I think it would be helpful to call out this is enabled via `VERSION 0.6`

Comment (dchw) [reply docs/earthfile/earthfile.md:683]: That was the point of the link; I was trying to not pull in so many details of how our feature flags work. I gave it a crack in the latest commit

Comment (dchw) [reply docs/earthfile/earthfile.md:683]: I added something but I will use your wording I guess

Comment (vladaionescu) [inline docs/earthfile/earthfile.md]: I think we need to reword this to just say "In VERSION 0.6...". Only advanced users / earthly contributors should know about individual feature flags.

File: docs/earthfile/earthfile.md [hunk 2]
```diff
@@ -677,6 +677,14 @@ SAVE IMAGE my-example-registry.com/another-image:latest
...
+
+This is automatically enabled when setting [`VERSION 0.6`](#version) in your `Earthfile`.
+{% endhint %}
```
Comment (vladaionescu) [inline docs/earthfile/earthfile.md:685]: Note that this applies to `SAVE ARTIFACT ... AS LOCAL ...` too. Should there be a similar warning there?

File: docs/earthfile/earthfile.md [hunk 3]
```diff
@@ -677,6 +677,13 @@ SAVE IMAGE my-example-registry.com/another-image:latest
 ```
 {% endhint %}
 
+{% hint style='danger' %}
+##### Important
+
+As of [`VERSION 0.6`](#version), the `--referenced-save-only` feature flag is enabled by default. Images are only saved if they are connected to the initial target through a chain of `BUILD` commands. [See the `--referenced-save-only` flag documentation for more details](./features.md#referenced-save-only).
```
Comment (vladaionescu) [inline docs/earthfile/earthfile.md:683]: Instead of the feature flag link here, I would link to BUILD and point to "What is output and pushed".

File: docs/earthfile/earthfile.md (hunk 1 above)
Comment (dchw) [reply docs/earthfile/earthfile.md]: See latest commit

File: docs/earthfile/earthfile.md (hunk 3 above)
Comment (dchw) [reply docs/earthfile/earthfile.md:683]: Why not both

File: docs/earthfile/earthfile.md (hunk 2 above)
Comment (dchw) [reply docs/earthfile/earthfile.md:685]: Yes, probably

File: docs/earthfile/earthfile.md (hunk 3 above)
Comment (vladaionescu) [reply docs/earthfile/earthfile.md:683]: I'm trying to avoid confusing new users with the term "feature flag" :-)

Just thinking that simpler is better - they don't need to know about our history to get going.

Comment (vladaionescu) [reply docs/earthfile/earthfile.md]: Friendly ping on this. I know it seems innocuous, but the extra simplicity is worth it.

**Review by vladaionescu — APPROVED** [review]

**Review by alexcb — APPROVED** [review]
